packages/library/src/send-transaction.ts

```diff
@@ -0,0 +1,200 @@
+            // transaction is lower. Cap the simulation commitment level to the level of the
+            // confirmation commitment.
+            preflightCommitment: commitment,
+        };
+    }
+    // The commitment at which the developer wishes to confirm the transaction is at least as
+    // high as the commitment at which they want to simulate it. Honour the config as-is.
+    return config;
+}
+
+async function sendTransaction_INTERNAL({
+    abortSignal,
+    commitment,
+    rpc,
+    transaction,
+    sendTransactionConfig,
+}: {
+    abortSignal: AbortSignal;
+    commitment: Commitment;
+    rpc: Rpc<SendTransactionApi>;
+    transaction: BaseTransaction &
+        ITransactionWithFeePayer &
+        (ITransactionWithBlockhashLifetime | IDurableNonceTransaction);
+    sendTransactionConfig?: SendTransactionConfigWithoutEncoding;
+}) {
+    const base64EncodedWireTransaction = getBase64EncodedWireTransaction(transaction);
+    await rpc
+        .sendTransaction(base64EncodedWireTransaction, {
+            ...getSendTransactionConfigWithAdjustedPreflightCommitment(commitment, sendTransactionConfig),
+            encoding: 'base64',
+        })
+        .send({ abortSignal });
+}
+
+export function createDefaultDurableNonceTransactionSender({
+    rpc,
+    rpcSubscriptions,
+}: DurableNonceTransactionSenderConfig) {
+    const confirmDurableNonceTransaction = createDefaultDurableNonceTransactionConfirmer({
+        rpc,
+        rpcSubscriptions,
+    });
+    return async function sendDurableNonceTransaction(
+        transaction: BaseTransaction & ITransactionWithFeePayer & IDurableNonceTransaction & IFullySignedTransaction,
+        config: Omit<
+            SendTransactionConfig &
+                Readonly<{
+                    abortSignal: AbortSignal;
+                    commitment: Commitment;
+                }>,
+            'encoding'
+        >
+    ): Promise<void> {
+        await sendAndConfirmDurableNonceTransaction({
+            ...config,
+            confirmDurableNonceTransaction,
+            rpc,
+            transaction,
+        });
+    };
+}
+
+export function createDefaultTransactionSender({ rpc, rpcSubscriptions }: RecentTransactionSenderConfig) {
+    const confirmRecentTransaction = createDefaultRecentTransactionConfirmer({
+        rpc,
+        rpcSubscriptions,
+    });
+    return async function sendTransaction(
+        transaction: SendableTransaction & ITransactionWithBlockhashLifetime,
+        config: Omit<
+            SendTransactionConfig &
+                Readonly<{
+                    abortSignal: AbortSignal;
+                    commitment: Commitment;
+                }>,
+            'encoding'
+        >
+    ): Promise<void> {
+        await sendAndConfirmTransaction({
+            ...config,
+            confirmRecentTransaction,
+            rpc,
+            transaction,
+        });
+    };
+}
+
+export async function sendAndConfirmDurableNonceTransaction({
```

Comment:
By splitting this into two methods an optimizing compiler can shake away all the durable nonce stuff if your app doesn't end up using it.